cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -400,6 +400,10 @@ func createDevice(device *v1alpha1.Device) types.Device {
 	for i, dtwin := range device.Status.Twins {
 		expected := &types.TwinValue{}
 		expected.Value = &device.Status.Twins[i].Desired.Value
+		metadataType, ok := device.Status.Twins[i].Desired.Metadata["type"]
```

Comment:
Does CRD definition support "Type" as well ??

Reply:
no we are using a map there.
https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/devicecontroller/apis/devices/v1alpha1/device_instance_types.go#L145

Reply:
No the CRD definition is generic and accepts map[string]string 
But we given an examples in sample crds which mentions the same https://github.com/kubeedge/kubeedge/blob/1a0d4e230e23036fc584991a95f06c968422e268/build/crd-samples/devices/create-device-instance.yaml#L34

Reply:
OK

Reply:
We need to mention all these points in usage docs. It can be done in future PRs

Reply:
+1